code/modules/paperwork/clipboard.dm

```diff
@@ -8,119 +11,167 @@
+		Paper.forceMove(user.loc)
+		user.put_in_hands(Paper)
+		to_chat(user, "<span class='notice'>You remove [Paper] from [src].</span>")
+		if(Paper == toppaper)
+			toppaper = null
+			var/obj/item/paper/newtop = locate(/obj/item/paper) in src
+			if(newtop && (newtop != Paper))
+				toppaper = newtop
+			else
+				toppaper = null
+		update_icon()
+
+/obj/item/clipboard/proc/remove_pen(mob/user)
+	pen.forceMove(user.loc)
+	user.put_in_hands(pen)
+	to_chat(user, "<span class='notice'>You remove [pen] from [src].</span>")
+	pen = null
+	update_icon()
+
+/obj/item/clipboard/AltClick(mob/user)
+	..()
+	if(toppaper)
+		remove_paper(toppaper, user)
+	else
+		remove_pen(user)
+
+/obj/item/clipboard/CtrlClick(mob/user)
+	..()
+	remove_pen(user)
+
 /obj/item/clipboard/update_overlays()
 	. = ..()
 	if(toppaper)
 		. += toppaper.icon_state
 		. += toppaper.overlays
-	if(haspen)
+	if(pen)
 		. += "clipboard_pen"
 	. += "clipboard_over"
 
 /obj/item/clipboard/attackby(obj/item/W, mob/user, params)
 	if(istype(W, /obj/item/paper))
+		//Add paper into the clipboard
 		if(!user.transferItemToLoc(W, src))
 			return
 		toppaper = W
-		to_chat(user, "<span class='notice'>You clip the paper onto \the [src].</span>")
-		update_appearance()
+		to_chat(user, "<span class='notice'>You clip [W] onto [src].</span>")
+	else if(istype(W, /obj/item/pen) && !pen)
+		//Add a pen into the clipboard, attack (write) if there is already one
+		if(!usr.transferItemToLoc(W, src))
+			return
+		pen = W
+		to_chat(usr, "<span class='notice'>You slot [W] into [src].</span>")
 	else if(toppaper)
 		toppaper.attackby(user.get_active_held_item(), user)
-		update_appearance()
-
+	update_appearance()
 
 /obj/item/clipboard/attack_self(mob/user)
-	var/dat = "<title>Clipboard</title>"
-	if(haspen)
-		dat += "<A href='?src=[REF(src)];pen=1'>Remove Pen</A><BR><HR>"
-	else
-		dat += "<A href='?src=[REF(src)];addpen=1'>Add Pen</A><BR><HR>"
-
-	//The topmost paper. You can't organise contents directly in byond, so this is what we're stuck with. -Pete
-	if(toppaper)
-		var/obj/item/paper/P = toppaper
-		dat += "<A href='?src=[REF(src)];write=[REF(P)]'>Write</A> <A href='?src=[REF(src)];remove=[REF(P)]'>Remove</A> - <A href='?src=[REF(src)];read=[REF(P)]'>[P.name]</A><BR><HR>"
-
-		for(P in src)
-			if(P == toppaper)
-				continue
-			dat += "<A href='?src=[REF(src)];write=[REF(P)]'>Write</A> <A href='?src=[REF(src)];remove=[REF(P)]'>Remove</A> <A href='?src=[REF(src)];top=[REF(P)]'>Move to top</A> - <A href='?src=[REF(src)];read=[REF(P)]'>[P.name]</A><BR>"
-	user << browse(dat, "window=clipboard")
-	onclose(user, "clipboard")
 	add_fingerprint(usr)
+	ui_interact(user)
+	return
+
+/obj/item/clipboard/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Clipboard")
+		ui.open()
+
+/obj/item/clipboard/ui_data(mob/user)
+	// prepare data for TGUI
+	var/list/data = list()
+	data["pen"] = "[pen]"
+
+	var/obj/item/paper/P = toppaper
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names